wallet/wallet.go

```diff
@@ -2464,6 +2464,69 @@ func (w *Wallet) GetTransactions(startBlock, endBlock *BlockIdentifier,
 	return &res, err
 }
 
+// GetTransactionResult returns a transaction in the UnminedTransaction
+// field if it is unmined, or in the MinedTransaction field as a Block
+// structure for a mined transaction.
+type GetTransactionResult struct {
```

Comment:
Just thinking about the use case of this method, it's easier if we just give a flat result back, something like,
```go
type GetTransactionResult struct {
	TransactionSummary

	// Height...
	Height int32

	// BlockHash...
	BlockHash chainhash.Hash
}
```
Callers can check the `Height` field to decide whether it's mined or not, or, if we wanna be explicit, we can add one more field, something like,

```go
type GetTransactionResult struct {
	TransactionSummary

	// Height...
	Height int32

	// BlockHash...
	BlockHash chainhash.Hash

	// Confirmations...
	Confirmations uint32
```

What do you think? This way we can also get rid of the field `Transactions: []TransactionSummary{summary}` which can be confusing.

Reply:
The rationale was to reuse the same importing technique as ListTransactionDetails uses in lnd (https://github.com/lightningnetwork/lnd/blob/91d910872d9a5d1490026a3a9f92a7afaa9af069/lnwallet/btcwallet/btcwallet.go#L1450).

Reply:
Sounds good. I'll provide an updated pr when I am done with the new tests.